airbyte-protocol/protocol-models/src/main/java/io/airbyte/protocol/models/CatalogHelpers.java

```diff
@@ -83,6 +83,22 @@ public static AirbyteCatalog configuredCatalogToCatalog(final ConfiguredAirbyteC
             .toList());
   }
 
+  /**
+   * Converts a {@link ConfiguredAirbyteCatalog} into an {@link AirbyteCatalog}. This is possible
+   * because the latter is a subset of the former. It filters out the non incremental streams.
+   *
+   * @param configuredCatalog - catalog to convert
+   * @return - airbyte catalog
+   */
+  public static AirbyteCatalog configuredCatalogToCatalogOnlyIncremental(final ConfiguredAirbyteCatalog configuredCatalog) {
```

Comment:
This method is a little bit weird to me, because an AirbyteCatalog with only the incremental-configured streams is not really useful anywhere - the only thing we ever really care about in that object is the stream descriptors of those streams. 

What if instead, this PR just added the `extractIncrementalStreamDescriptors()` method which looked like this:
```
  public static List<StreamDescriptor> extractIncrementalStreamDescriptors(final ConfiguredAirbyteCatalog configuredCatalog) {
    return configuredCatalog.getStreams().stream()
        .filter(streamAndConfig -> streamAndConfig.getSyncMode() == SyncMode.INCREMENTAL)
        .map(streamAndConfig -> extractDescriptor(streamAndConfig.getStream()))
        .toList();
  }
```

This way, this method pulls exactly what it needs out of the configured catalog without producing a strange intermediate catalog object that only contains some streams

Reply:
Make senses, done.